src/javaharness/java/arcs/api/PECInnerPortImpl.java

```diff
@@ -79,6 +80,7 @@ public void processMessage(PortableJson message) {
     PortableJson messageBody = message.getObject(MESSAGE_BODY_FIELD);
     switch (messageType) {
       case INSTANTIATE_PARTICLE_MSG:
+      case REINSTANTIATE_PARTICLE_MSG:
```

Comment:
Are these two really supposed to run the exact same code?

Reply:
except the `if (mapper.hasThingForIdentifier(particleId)) {` statement on ln 93 - for REINSTANTIATE it'll always be true (reinstantiation only happens for non-factory instantiated particles, when particle is already mapped) - added an assert to verify this